src/Storage/Adapter/Memory.php

```diff
@@ -168,22 +170,43 @@ public function increment(string $key, int $value = 1)
     }
 
     /**
-     * Stores data in the adapter
+     * Stores data in the adapter. If the TTL is `null` (default) or not defined
+     * then the default TTL will be used, as set in this adapter. If the TTL
+     * is `0` or a negative number, a `delete()` will be issued, since this
+     * item has expired. If you need to set this key forever, you should use
```

Comment:
And what if `$ttl instanceof DateInterval`?

Reply:
If it is `DateInterval` it cannot be 0 and as such it will not expire instantly. If it is `DateInterval` and compared to `time()` it has expired, then it will be deleted.

Reply:
But it is not `is_int()`.

Reply:
It doesn't matter for this adapter since it is the Memory one. The data will be wiped after the request is completed.